tfhe/src/high_level_api/keys/public.rs

```diff
@@ -189,3 +191,54 @@ macro_rules! impl_ref_key_from_compressed_public_keychain {
         }
     }
 }
+
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
+pub struct CompactPublicKey {
+    #[cfg(feature = "integer")]
+    pub(in crate::high_level_api) integer_key: IntegerCompactPublicKey,
+}
+
+impl CompactPublicKey {
+    pub fn new(client_key: &ClientKey) -> Self {
+        #[cfg(feature = "integer")]
+        {
+            Self {
+                integer_key: IntegerCompactPublicKey::new(&client_key.integer_key),
+            }
+        }
+        #[cfg(not(feature = "integer"))]
+        {
+            let _ = client_key;
+            Self {}
+        }
+    }
+}
+
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
+pub struct CompressedCompactPublicKey {
+    #[cfg(feature = "integer")]
+    pub(in crate::high_level_api) integer_key: IntegerCompressedCompactPublicKey,
+}
+
+impl CompressedCompactPublicKey {
+    pub fn new(client_key: &ClientKey) -> Self {
+        #[cfg(feature = "integer")]
+        {
+            Self {
+                integer_key: IntegerCompressedCompactPublicKey::new(&client_key.integer_key),
+            }
+        }
+        #[cfg(not(feature = "integer"))]
+        {
+            let _ = client_key;
+            Self {}
+        }
+    }
+
+    pub fn decompress(self) -> CompactPublicKey {
+        CompactPublicKey {
+            #[cfg(feature = "integer")]
+            integer_key: self.integer_key.decompress(),
+        }
+    }
```

Comment:
btw I know it was a debate at some point about taking the entity by ref or by value, we can change it if you want to take stuff by ref

Reply:
Usually unless i'm reusing the memory from the self object I use &self not self

Reply:
Its propably better to wait another time, its not a huge blocker/problem

Reply:
agreed

Reply:
I'll check we have a backlog issue

Reply:
https://github.com/zama-ai/tfhe-rs-internal/issues/147